src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const currentSum = sum;
+
+      sum = 0;
+
+      return currentSum;
```

Comment:
The check for `undefined` is correct to determine when to return the sum and reset it. However, ensure that the function's behavior is well-documented so that users of this function understand that passing no arguments triggers the calculation and reset of the sum.